### PR TITLE
Implement container registry for a dataset

### PR DESCRIPTION
**Pull request by illusional**
Will be useful for a universially dependent `cpg-common` dataset with an artifact registry, eventually removing the support for specifying cpg-common manually.

\+ Import the pre-existing fewgenomes repository.

### CODE REVIEW

File: cpg_infra/driver.py [hunk 1]
```diff
@@ -960,7 +960,40 @@ def setup_container_registry(self):
...
+                registry=custom_container_registry,
+                member=account,
+                membership=ContainerRegistryMembership.READER,
+            )
+            if kind in ('standard', 'full'):
+                self.infra.add_member_to_container_registry(
+                    f'{kind}-images-writer-in-container-registry',
+                    registry=custom_container_registry,
+                    member=account,
+                    membership=ContainerRegistryMembership.APPEND,
```
Comment (lgruen) [inline cpg_infra/driver.py:988]: Not directly related to this PR, but I just noted that `ContainerRegistryMembership.APPEND` is implemented as `role = 'roles/artifactregistry.writer'`, which seems to imply more than just appending? (I.e. also overwrite + delete?)
https://github.com/populationgenomics/cpg-infrastructure/blob/97d72179727f27a991a318ffae3e7b30990fe92c/cpg_infra/abstraction/gcp.py#L487-L488

This seems different from what we typically allow for `standard` (although we might want to revise that anyhow).

File: cpg_infra/driver.py [hunk 2]
```diff
@@ -960,7 +960,40 @@ def setup_container_registry(self):
...
+                )
+
+    def setup_inherited_container_registries(self):
```
Comment (lgruen) [inline cpg_infra/driver.py:991]: If this is mostly meant to handle our existing setup temporarily, maybe "legacy" is the better description?

```suggestion
    def setup_legacy_container_registries(self):
```

If you want to change the implementation later to support access to repositories of dataset dependencies, maybe `dependent` could be a better description?

File: cpg_infra/driver.py [hunk 3]
```diff
@@ -991,14 +1024,14 @@ def setup_container_registry(self):
...
-                    member=account,
-                    membership=ContainerRegistryMembership.APPEND,
-                )
+        for kind in ('full', 'standard'):
+            self.infra.add_member_to_container_registry(
+                f'{kind}-images-writer-in-cpg-common',
+                registry=self.config.gcp.common_artifact_registry_name,
+                project=self.config.gcp.common_artifact_registry_project,
+                member=self.access_level_groups[kind],
+                membership=ContainerRegistryMembership.APPEND,
```
Comment (lgruen) [inline cpg_infra/driver.py:1033]: Ditto.

File: stack/Pulumi.fewgenomes.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@ config:
   datasets:gcp_hail_service_account_full: fewgenomes-full-915@hail-295901.iam.gserviceaccount.com
   datasets:gcp_hail_service_account_standard: fewgenomes-standard-737@hail-295901.iam.gserviceaccount.com
   datasets:gcp_hail_service_account_test: fewgenomes-test-126@hail-295901.iam.gserviceaccount.com
+  datasets:create_container_registry: "true"
```
Comment (lgruen) [inline stack/Pulumi.fewgenomes.yaml:8]: Or maybe remove this if you want to delete the registry anyhow? You might have to delete all images manually first.

**Review by lgruen — APPROVED** [review]

File: cpg_infra/driver.py (hunk 1 above)
Comment (illusional) [reply cpg_infra/driver.py]: Agreed, renamed to `writer`, although I believe you can't overwrite (as no delete) with writer: https://cloud.google.com/artifact-registry/docs/access-control

File: cpg_infra/driver.py (hunk 2 above)
Comment (illusional) [reply cpg_infra/driver.py]: Yep agreed, renamed to legacy.

File: stack/Pulumi.fewgenomes.yaml (hunk 4 above)
Comment (illusional) [reply stack/Pulumi.fewgenomes.yaml]: Removed and deleted the artifact repos

**Pull request merged**